modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/UpdateCollectionDisplayConfigMVCActionCommand.java

```diff
@@ -107,16 +107,12 @@ protected void doProcessAction(
 				FragmentEntryProcessorConstants.
 					KEY_FREEMARKER_FRAGMENT_ENTRY_PROCESSOR);
 
-			if ((configuration == null) || !JSONUtil.isValid(configuration)) {
-				continue;
-			}
+			JSONObject configurationJSONObject = JSONUtil.createJSONObject(
```

Comment:
@slnn You did not check (configuration == null), there may be an NPE after your change.

Reply:
@tinatian Actually the null protection is not needed. In the new PR the null check is removed in a separate commit to make it clear: https://github.com/liferay-core-infra/liferay-portal/pull/1101/commits/8f41ce2502d5ad2e90f707564951294eb6f8586d